### PR TITLE
feat!: add igor faucet

### PR DESCRIPTION
**Pull request by hansieodendaal**
Description
---
- Updated the igor genesis block to contain a faucet with revealed values
- Added a new range-proof option for the genesis block sanity test

Motivation and Context
---
- Evaluating faucet with revealed values

How Has This Been Tested?
---
- Unit test demonstrating the range proof for each output in the faucet
- Performed system-level testing to demonstrate that the igor blockchain works properly - 2x base nodes, 2x wallets, 1x SHA3 miner, 1x merge mining proxy, 1x XMRig miner.

What process can a PR reviewer use to test or verify this change?
---
Perform system-level testing

Breaking Changes
---

- [ ] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [x] Other - New genesis block.

 BREAKING CHANGE: The chain has to be restarted - delete the blockchain database and resync the chain

### CODE REVIEW

File: base_layer/core/src/consensus/consensus_constants.rs [hunk 1]
```diff
@@ -428,17 +428,18 @@ impl ConsensusConstants {
     pub fn igor() -> Vec<Self> {
         let mut algos = HashMap::new();
         // sha3/monero to 40/60 split
+        const MONERO_TARGET_TIME: u64 = 10; // usually 200 for a 120 average block time
```
Comment (SWvheerden) [inline base_layer/core/src/consensus/consensus_constants.rs]: make it two variables, its much more clear, one for monero, one for sha3
Its going to clear up the math below quite a bit

File: base_layer/core/src/consensus/consensus_constants.rs [hunk 2]
```diff
@@ -428,17 +428,18 @@ impl ConsensusConstants {
     pub fn igor() -> Vec<Self> {
         let mut algos = HashMap::new();
         // sha3/monero to 40/60 split
```
Comment (SWvheerden) [inline base_layer/core/src/consensus/consensus_constants.rs]:
```suggestion
        // sha3/monero to 25/75 split
```
thats the correct split here
avg block time = (10+15) / 2 = 7.5
thus percentage monero block = 7.5/x = 10, then 75%. 
Thus sha3 = 25%

Formula for calculating the numbers here is specified in `struct PowAlgorthimConstants`
Target time = {avg block time}/{Desried block percentage} = {algo target time}

Comment (SWvheerden) [reply base_layer/core/src/consensus/consensus_constants.rs]: Perhaps we should change this to make it correct 40/60

File: base_layer/core/src/blocks/genesis_block.rs [hunk 3]
```diff
@@ -81,8 +80,8 @@ pub fn get_stagenet_genesis_block() -> ChainBlock {
 }
 
 fn get_stagenet_genesis_block_raw() -> Block {
-    // Note: Use print_new_genesis_block_nextnet in core/tests/helpers/block_builders.rs to generate the required
-    // fields below
+    // Note: Use 'print_new_genesis_block_stagenet' in core/tests/helpers/block_builders.rs to generate the required
+    // fields below - DO THIS ONCE ONLY AS IT CONTAINS RANDOMNESS
```
Comment (SWvheerden) [inline base_layer/core/src/blocks/genesis_block.rs]:
```suggestion
    // Note: Use 'print_new_genesis_block_stagenet' in core/tests/helpers/block_builders.rs to generate a new block an get the required
    // fields below - DO THIS ONCE ONLY AS IT GENERATES NEW RANDOM FIELDS
```

**Review by SWvheerden — COMMENTED** [review]
Looks good, just have a few nits

File: base_layer/core/src/consensus/consensus_constants.rs (hunk 2 above)
Comment (hansieodendaal) [reply base_layer/core/src/consensus/consensus_constants.rs]: It should be 40/60

Comment (hansieodendaal) [reply base_layer/core/src/consensus/consensus_constants.rs]: Updated as:
```rust
        // For SHA3/Monero to have a 40/60 split:
        // sha3_target_time = monero_target_time * (100 - 40) / 40
        // monero_target_time = sha3_target_time * (100 - 60) / 60
        // target_time = monero_target_time * sha3_target_time / (monero_target_time + sha3_target_time)
        const MONERO_SPLIT: u64 = 60;                             // Must be < 100
        const SHA3_SPLIT: u64 = 100 - MONERO_SPLIT;
        const MONERO_TARGET_TIME: u64 = 20;                       // usually 200 (for a 120 average block time)
        const SHA3_TARGET_TIME: u64 = MONERO_TARGET_TIME * (100 - SHA3_SPLIT) / SHA3_SPLIT;
        algos.insert(PowAlgorithm::Sha3, PowAlgorithmConstants {
            max_target_time: SHA3_TARGET_TIME * 6,                // target_time x 6
            min_difficulty: (SHA3_TARGET_TIME * 200_000).into(),  // target_time x 200_000
            max_difficulty: u64::MAX.into(),
            target_time: SHA3_TARGET_TIME,                        // (monero_target_time x 1.5)
        });
        algos.insert(PowAlgorithm::Monero, PowAlgorithmConstants {
            max_target_time: MONERO_TARGET_TIME * 6,           // target_time x 6
            min_difficulty: (MONERO_TARGET_TIME * 300).into(), // target_time x 300
            max_difficulty: u64::MAX.into(),
            target_time: MONERO_TARGET_TIME,
        });
```

**Review by SWvheerden — APPROVED** [review]

**Pull request merged**